### DISCUSSION TITLE
gRPC request through bash

### DISCUSSION THREAD

**Discussion by beatrizgoa** [triton-inference-server/server#7407] [Q&A]
Hi,

I'd like to do a gRPC request to triton server using the bash and jmeter, but I am having issues.

I have started my triton serves as:

```
docker run --shm-size=1g --rm -p 8000:8000 -p 8001:8001 -p 8002:8002 -v /home/models_rep:/models nvcr.io/nvidia/tritonserver:23.06-py3 tritonserver --model-repository=/models --repository-poll-secs=10 --model-control-mode=poll
```
and I am trying to do the request by bash like:

```
grpcurl -d @ -plaintext localhost:8001 inference.GRPCInferenceService.ModelInfer < load_grcp.json

```
but i am having the following error:

```
Error invoking method "inference.GRPCInferenceService.ModelInfer": failed to query for service descriptor "inference.GRPCInferenceService": server does not support the reflection API
```

I think that the method I am trying to use does not exist. I have been reading and maybe I need to generate a proto file, I have tried to do that from [here](https://github.com/triton-inference-server/common/tree/main/protobuf) like:

```
protoc --descriptor_set_out=grpc_service_descriptor.pb --include_imports common/protobuf/grpc_servi
ce.proto
```
And I have obtained:
```
model_config.proto: File not found.
common/protobuf/grpc_service.proto:33:1: Import "model_config.proto" was not found or had errors.
common/protobuf/grpc_service.proto:873:3: "ModelConfig" is not defined.
```

And I dont know how should I proceed to generate the bash request to be able then do it from jmeter

Thanks a lot for your help :)

**Comment by jhabaa** [comment]
Hi, 
Did you found a solution ?

**Comment by Madihaa-Shaikh** [comment]
hi , did you find the solution?
